lib/eventFactory.js

```diff
@@ -65,20 +70,29 @@ function getJobsFromJobName(config) {
  * @param  {Object}   config
  * @param  {Array}    config.jobs                           Array of job objects
  * @param  {Number}   config.prNum                          PR number
+ * @param  {Object}   config.pipelineConfig
+ * @param  {Object}   config.pipelineConfig.workflowGraph   Object with nodes and edges that represent the order of jobs
  * @param  {String}   config.startFrom                      Startfrom (e.g. ~commit, ~pr, etc)
  * @resolves {Array}                                        Array of PR jobs to start
  */
 function getJobsFromPR(config) {
-    const { jobs, prNum, startFrom } = config;
+    const { jobs, prNum, pipelineConfig, startFrom } = config;
+    const isPRTrigger = PR_TRIGGER.test(startFrom);
 
-    if (startFrom !== '~pr') {
+    if (!isPRTrigger) {
         return [];
     }
 
-    const PRJobArray = jobs.filter(j => j.name.startsWith(`PR-${prNum}:`));
-    const jobsToStart = PRJobArray.filter(j => j.state === 'ENABLED' && !j.archived);
+    const nextJobs = workflowParser.getNextJobs(pipelineConfig.workflowGraph, {
+        trigger: startFrom,
+        prNum
+    });
 
-    return jobsToStart;
+    const jobsToStart = jobs.filter(
+        j => j.name.startsWith(`PR-${prNum}:`) || nextJobs.includes(j.name)
```

Comment:
Do you still need to do the additional filter to make sure the jobs are ENABLED and not archived (`PRJobArray.filter(j => j.state === 'ENABLED' && !j.archived);`)?